source/postprocess/visualization.cc

```diff
@@ -531,46 +531,49 @@ namespace aspect
 
       int tmp_file_desc = -1;
 
-      {
-        // Try getting the environment variable for the temporary directory
-        const char *tmp_filedir = getenv("TMPDIR");
-        // If we can't, default to /tmp
-        if (tmp_filedir)
-          tmp_filename = tmp_filedir;
-        else
-          tmp_filename = "/tmp";
-        tmp_filename += "/aspect.tmp.XXXXXX";
-
-        // Create the temporary file; get at the actual filename
-        // by using a C-style string that mkstemp will then overwrite
-        char *tmp_filename_x = new char[tmp_filename.size()+1];
-        std::strcpy(tmp_filename_x, tmp_filename.c_str());
-        tmp_file_desc = mkstemp(tmp_filename_x);
-        tmp_filename = tmp_filename_x;
-        delete []tmp_filename_x;
-
-        // If we failed to create the temp file, just write directly to the target file.
-        // We also provide a warning about this fact. There are places where
-        // this fails *on every node*, so we will get a lot of warning messages
-        // into the output; in these cases, just writing multiple pieces to
-        // std::cerr will produce an unreadable mass of text; rather, first
-        // assemble the error message completely, and then output it atomically
-        if (tmp_file_desc == -1)
-          {
-            const std::string x = ("***** WARNING: could not create temporary file <"
-                                   +
-                                   tmp_filename
-                                   +
-                                   ">, will output directly to final location. This may negatively "
-                                   "affect performance. (On processor "
-                                   + Utilities::int_to_string(Utilities::MPI::this_mpi_process (MPI_COMM_WORLD))
-                                   + ".)\n");
-
-            std::cerr << x << std::flush;
-
-            tmp_filename = *filename;
-          }
-      }
+      if (system(0) != 0)
```

Comment:
So is this is how this works on the BG/Q? The `system` function is declared in header files, but it will always yield an error?

I find the call `system(0)` odd to understand. Maybe write it as `system((char*)0)`?